src/main/java/org/folio/circulation/domain/RequestQueue.java

```diff
@@ -48,18 +49,17 @@ public boolean isRequestedByOtherPatron(User requestingUser) {
   }
 
   public void add(Request newRequest) {
-    // NOTE: assumes Collection supports add
+    requests = new ArrayList<>(requests);
     requests.add(newRequest);
     orderRequests();
   }
 
   public void remove(Request request) {
-    // NOTE: tests use Arrays.asList which produces a collection which
-    // does not support remove
     requests = requests.stream()
       .filter(r -> !r.getId().equals(request.getId()))
       .collect(Collectors.toList());
     request.removePosition();
+    request.freePreviousPosition();
```

Comment:
What is the value in rem Bering the previous position on removal, when it is forgotten immediately afterwards?

Reply:
There is no value. Forgot to remove it. Thanks.